File: packages/client_react/src/types/auditLog.ts
export interface AuditLogDto {
  id: number;
  tableName: string;
  recordKeys: Record<string, any>;
  action: string;
  oldData: string | null;
  newData: string | null;
  changedFields: string[] | null;
  userId: number;
  userTag: string;
  reasonForChange: string | null;
  timestamp: string;
  clientIp: string | null;
}

export interface PaginatedAuditLogsDto {
  auditLogs: AuditLogDto[];
  totalCount: number;
  pageSize: number;
  currentPage: number;
}

File: server/QCLab/Models/AuditLog.cs
using System;
using System.Collections.Generic;
using System.Text.Json;

namespace QCLab.Models;

public partial class AuditLog
{
    public long Id { get; set; }

    public string TableName { get; set; } = null!;

    public JsonElement RecordKeys { get; set; }

    public string Action { get; set; } = null!;

    public string? OldData { get; set; }

    public string? NewData { get; set; }

    public List<string>? ChangedFields { get; set; }

    public long UserId { get; set; }

    public string? ReasonForChange { get; set; }

    public DateTime Timestamp { get; set; }

    public string? ClientIp { get; set; }

    public virtual User User { get; set; } = null!;
}
